src/main/java/com/github/coleb1911/ghost2/commands/modules/fun/ModuleFortune.java

```diff
@@ -0,0 +1,94 @@
+ * @author John Allison | john123allison@gmail.com
+ */
+public final class ModuleFortune extends Module {
+    private static final Random random = new Random();
+    private final String API_URL = " http://yerkee.com/api/fortune/";
+    private final String[] FORTUNE_CATEGORIES = {"all", "bible", "computers", "cookie", "definitions", "miscellaneous",
+            "people", "platitudes", "politics", "science", "wisdom"};
+
+    @ReflectiveAccess
+    public ModuleFortune() {
+        super(new ModuleInfo.Builder(ModuleFortune.class)
+                .withName("fortune")
+                .withDescription("Fetch a free fortune!"));
+    }
+    
+    @Override
+    @ReflectiveAccess
+    public void invoke(@NotNull CommandContext ctx) throws IOException {
+        String category;
+        String categorySelected;
+
+        // check if args are provided - if not, assign a random category to the fortune
+        // if an invalid arg is provided, return a help message
+        if (!ctx.getArgs().isEmpty()) {
+            categorySelected  = ctx.getArgs().get(0);
+            if (Arrays.asList(FORTUNE_CATEGORIES).contains(categorySelected)) {
+                category = categorySelected;
+            } else {
+                // This looks weird because had to apply the fencepost problem to format this correctly
+                String reply = "Invalid argument. List of available categories: " + FORTUNE_CATEGORIES[0] + " ";
+                for(int i = 1; i < FORTUNE_CATEGORIES.length - 1; i++) {
+                    reply += ", " + FORTUNE_CATEGORIES[i];
+                }
+                ctx.replyBlocking(reply);
+                return;
+            }
+        } else {
+            int index = random.nextInt(FORTUNE_CATEGORIES.length - 1);
+            category = FORTUNE_CATEGORIES[index];
+        }
+
+        // make a url and cast it as a HttpURLConnection
+        String idURL = API_URL + category;
+        URL url = new URL(idURL);
+        HttpURLConnection conn = (HttpURLConnection)url.openConnection();
+
+        conn.setRequestMethod("GET");
+        conn.connect();
+
+        int responseCode = conn.getResponseCode();
+
+        // read response
+        String json = "";
+        if(responseCode != 200) {
+            throw new RuntimeException("Response Code" + responseCode);
+        } else {
+            Scanner scanner = new Scanner(url.openStream());
+            while(scanner.hasNext()) {
+                json += scanner.nextLine();
+            }
+
+            scanner.close();
+
+            // parse JSON
+            try {
+                JSONObject obj = new JSONObject(json);
+                String fortune = (String) obj.get("fortune");
+                ctx.replyBlocking(fortune);
+            } catch (JSONException e) {
+                System.out.println("JSON parsing Exception");
+            }
+        }
+    }
+}
```

Comment:
```suggestion
}

@JsonIgnoreProperties(ignoreUnknown = true)
private static class FortuneResult {
	@JsonProperty("fortune")
	private String fortune;
	
	String getFortune() {
		return fortune;
	}
} 
```

Reply:
Here's your starting point. I don't want to write too much of the code for you.

You'll need to obtain a `RestTemplate` instance as well using `RestUtils#defaultRestTemplate`. Pass the URL and `FortuneResult.class` into `RestTemplate#getForObject`. The finicky stuff is hidden behind that one `getForObject` call and you're free to read the result object as you wish.

Again, see the other Fun modules if you need an example!